src/main/java/com/ace/alfox/controllers/ActionController.java

```diff
@@ -1,35 +1,35 @@
-package com.ace.alfox;
-
-import com.ace.alfox.game.GameCharacter;
-import com.ace.alfox.lib.Action;
-import com.ace.alfox.lib.CharacterFactory;
-import com.ace.alfox.lib.Result;
-import org.springframework.http.*;
-import org.springframework.web.bind.annotation.*;
-
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpSession;
-
-import java.io.*;
-import java.util.Map;
-
-@RestController
-class ActionController {
-
-    @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
-    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
-        HttpSession session = request.getSession(true);
-        String characterId = (String) session.getAttribute("cid");
-
-        GameCharacter player = CharacterFactory.fetch(characterId);
-        if(player == null) {
-            return ResponseEntity.notFound().build();
-        }
-
-        Action doSomething = Action.named(command).withParameters(params);
-        Result result = player.applyAction(doSomething);
-        CharacterFactory.save(result.character);
-
-        return ResponseEntity.ok(result);
-    }
+package com.ace.alfox.controllers;
+
+import com.ace.alfox.game.models.Player;
+import com.ace.alfox.lib.ActionFactory;
+import com.ace.alfox.lib.ActionWebResult;
+import com.ace.alfox.lib.CharacterFactory;
+import org.springframework.http.*;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import java.io.*;
+import java.util.Map;
+
+@RestController
+class ActionController {
+
+    @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
+    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
+        HttpSession session = request.getSession(true);
+        String characterId = (String) session.getAttribute("cid");
```

Comment:
Might be useful to have a base class like `BaseApi` which can provide `getCharacterId(request)`, not a blocker. (It could also return Player directly since that seems to be why we would want the characterId).

Reply:
You're right, but I don't think we're going to have many (if any) other `HttpController`s with this ActionController acting as an internal "router". So I'm going to vote that we hold out hiding the tedium until it becomes tedious.